frontend/pages/shelter/[shelterId]/pet/[petId].tsx

```diff
@@ -9,76 +10,25 @@ import { useRouter } from "next/dist/client/router";
 import React, { useEffect, useState } from "react";
 import styled from "styled-components";
 
-// export const defaultPet: Animal.Attributes = {
-// 	key: "",
-// 	name: "Cat 1",
-// 	images: [
-// 		"https://via.placeholder.com/86",
-// 		"https://via.placeholder.com/86",
-// 		"https://via.placeholder.com/86",
-// 		"https://via.placeholder.com/86",
-// 		"https://via.placeholder.com/86",
-// 	],
-// 	visible: false,
-// 	species: Species.CAT,
-// 	status: Status.HEALTHY,
-// 	acquired: new Date(),
-// 	breed: "Shorthair cat",
-// 	sex: Sex.MALE,
-// 	furLength: FurLength.SHORT,
-// 	medicalIssues: ["asthma"],
-// 	sterilised: Sterilised.YES,
-// 	dateOfBirth: new Date(),
-// 	furColor: ["white", "brown"],
-// 	toiletTrained: true,
-// };
-
 export default function PetDetails() {
 	const router = useRouter();
 	const petId = router.query.petId as string;
 	const [petData, setPetData] = useState<Animal.Attributes>();
 	const [petAdopters, setPetAdopters] = useState<Adopter[]>();
+	const [loading, setLoading] = useState(true);
 
 	useEffect(() => {
-		/* TODO: Refactor to either use API from contract or receive data from parent */
-		console.log(`Fetching pet info for shelter - ${petId}`);
-		const pd: Partial<Animal.Attributes> = {
-			id: petId,
-			name: "Cat 1",
-			animalImages: [
-				{
-					photoUrl: "https://via.placeholder.com/86",
-					thumbnailUrl: "https://via.placeholder.com/86",
-				},
-				{
-					photoUrl: "https://via.placeholder.com/86",
-					thumbnailUrl: "https://via.placeholder.com/86",
-				},
-				{
-					photoUrl: "https://via.placeholder.com/86",
-					thumbnailUrl: "https://via.placeholder.com/86",
-				},
-				{
-					photoUrl: "https://via.placeholder.com/86",
-					thumbnailUrl: "https://via.placeholder.com/86",
-				},
-				{
-					photoUrl: "https://via.placeholder.com/86",
-					thumbnailUrl: "https://via.placeholder.com/86",
-				},
-				{
-					photoUrl: "https://via.placeholder.com/86",
-					thumbnailUrl: "https://via.placeholder.com/86",
-				},
-			],
-			visible: false,
-			species: Animal.Species.Cat,
-			adoptionStatus: Animal.AdoptionStatus.Healthy,
-			intakeDate: new Date(),
-			breed: "Shorthair cat",
+		const fetchPetDetail = async () => {
+			try {
+				const res = await new PetApi().fetchPetData(petId);
```

Comment:
`petId` is initially undefined ([oof](https://github.com/vercel/next.js/discussions/11484)), add a check for it to be defined before calling the api